arrows/mvg/applets/init_cameras_landmarks.cxx

```diff
@@ -0,0 +1,513 @@
+    initialize_cameras_landmarks::set_nested_algo_configuration( "initializer",
+                                                                 config,
+                                                                 algorithm );
+  }
+
+  void clear_ptrs()
+  {
+    camera_map_ptr = nullptr;
+    landmark_map_ptr = nullptr;
+    feature_track_set_ptr = nullptr;
+    sfm_constraint_ptr = nullptr;
+  }
+
+  void load_tracks( )
+  {
+    kwiver::vital::path_t in_tracks_path =
+      config->get_value<kwiver::vital::path_t>("input_tracks_file");
+    feature_track_set_ptr =
+      kwiver::vital::read_feature_track_file(in_tracks_path);
+  }
+
+  void load_sfm_constraint( )
+  {
+    if(config == nullptr)
+    {
+      return;
+    }
+
+    sfm_constraint_ptr = std::make_shared<sfm_constraints>();
+    if(config->has_value("video_source") &&
+       config->get_value<std::string>("video_source") != "")
+    {
+      video_input_sptr video_reader;
+      video_input::set_nested_algo_configuration("video_reader", config,
+                                                 video_reader);
+      video_input::get_nested_algo_configuration("video_reader", config,
+                                                 video_reader);
+
+      video_reader->open( config->get_value<std::string>("video_source") );
+      if (video_reader->get_implementation_capabilities()
+        .has_capability(video_input::HAS_METADATA))
+      {
+        kwiver::vital::timestamp currentTimestamp;
+        kwiver::vital::metadata_map::map_metadata_t map_t;
+        while (video_reader->next_frame(currentTimestamp))
+        {
+          auto frame = currentTimestamp.get_frame();
+          auto mdVec = video_reader->frame_metadata();
+
+          if (mdVec.size() > 0)
+          {
+            map_t.emplace(frame, mdVec);
+          }
+        }
+        kwiver::vital::metadata_map_sptr metadata_map =
+        std::make_shared<kwiver::vital::simple_metadata_map>(map_t);
+        sfm_constraint_ptr->set_metadata(metadata_map);
+      }
+    }
+  }
+
+  bool write_cameras()
+  {
+    std::string output_cameras_director =
+    config->get_value<std::string>("output_cameras_director");
+    std::string camera_prefix = config->get_value<std::string>("camera_prefix");
+
+    for( auto iter: camera_map_ptr->cameras())
+    {
+      int fn = iter.first;
+      camera_sptr cam = iter.second;
+      std::string out_fname = output_cameras_director + "/" + camera_prefix +
+      std::to_string(fn) + ".krtd";
+      kwiver::vital::path_t out_path(out_fname);
+      auto cam_ptr = std::dynamic_pointer_cast<camera_perspective>(cam);
+      write_krtd_file( *cam_ptr, out_path );
+    }
+
+    return true;
+  }
+
+  bool write_landmarks()
+  {
+    kwiver::vital::path_t out_landmarks_path =
+      config->get_value<kwiver::vital::path_t>("output_landmarks_filename");
+
+    // verify that we can open the output file for writing
+    // so that we don't find a problem only after spending
```

Comment:
As the comment indicates, this check would be better run before running all the computation.  I suggest moving this file write access check to the `check_config` function that we error out earlier if the file path is not writable.